internal/stanza/frompdataconverter_test.go

```diff
@@ -0,0 +1,277 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stanza
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-log-collection/entry"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+)
+
+func BenchmarkConvertFromPdataSimple(b *testing.B) {
+	b.StopTimer()
+	pLogs := pdata.NewLogs()
+	b.StartTimer()
+
+	for i := 0; i < b.N; i++ {
+		ConvertFrom(pLogs)
+	}
+}
+
+func BenchmarkConvertFromPdataComplex(b *testing.B) {
+	b.StopTimer()
+	pLogs := complexPdataForNDifferentHosts(1, 1)
+	b.StartTimer()
+
+	for i := 0; i < b.N; i++ {
+		ConvertFrom(pLogs)
+	}
+}
+
+func baseMap() pdata.Map {
+	obj := pcommon.NewMap()
+	obj.InsertBool("bool", true)
+	obj.InsertInt("int", 123)
+	obj.InsertDouble("double", 12.34)
+	obj.InsertString("string", "hello")
+	obj.InsertBytes("bytes", []byte("asdf"))
+	return obj
+}
+
+func baseMapValue() pdata.Value {
+	v := pcommon.NewValueMap()
+	baseMap := baseMap()
+	baseMap.CopyTo(v.MapVal())
+	return v
+}
+
+func complexPdataForNDifferentHosts(count int, n int) pdata.Logs {
+	pLogs := pdata.NewLogs()
+	logs := pLogs.ResourceLogs()
+
+	for i := 0; i < count; i++ {
+		rls := logs.AppendEmpty()
+
+		resource := rls.Resource()
+		attr := baseMap()
+		attr.Insert("object", baseMapValue())
+		attr.InsertString("host", fmt.Sprintf("host-%d", i%n))
+		attr.CopyTo(resource.Attributes())
+
+		ills := rls.ScopeLogs().AppendEmpty()
+		lr := ills.LogRecords().AppendEmpty()
+
+		lr.SetSpanID(pcommon.NewSpanID([8]byte{0x32, 0xf0, 0xa2, 0x2b, 0x6a, 0x81, 0x2c, 0xff}))
+		lr.SetTraceID(pcommon.NewTraceID([16]byte{0x48, 0x01, 0x40, 0xf3, 0xd7, 0x70, 0xa5, 0xae, 0x32, 0xf0, 0xa2, 0x2b, 0x6a, 0x81, 0x2c, 0xff}))
+		lr.SetFlags(uint32(0x01))
+
+		lr.SetSeverityNumber(plog.SeverityNumberERROR)
+		lr.SetSeverityText("Error")
+
+		attr.Remove("double")
+		attr.Remove("host")
+		attr.CopyTo(lr.Attributes())
+
+		body := baseMapValue()
+		level2 := baseMapValue()
+		level2.MapVal().Remove("bytes")
+		level1 := baseMapValue()
+		level1.MapVal().Insert("object", level2)
+		body.MapVal().Insert("object", level1)
+		body.CopyTo(lr.Body())
+	}
+	return pLogs
+}
+
+func TestConvertFrom(t *testing.T) {
```

Comment:
A round trip test would be really useful here.

Reply:
Will work on expanding test coverage.